apiextensions/storageversion/migrator_test.go

```diff
@@ -88,6 +89,44 @@ func TestMigrate(t *testing.T) {
 	)
 }
 
+func TestMigrate_SingleStoredVersion(t *testing.T) {
+	singleVersionFakeCRD := &apix.CustomResourceDefinition{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: fakeGR.String(),
+		},
+		Spec: apix.CustomResourceDefinitionSpec{
+			Group: fakeGK.Group,
+			Versions: []apix.CustomResourceDefinitionVersion{
+				{Name: "v1", Served: true, Storage: true},
+			},
+		},
+		Status: apix.CustomResourceDefinitionStatus{
+			StoredVersions: []string{"v1"},
+		},
+	}
+
+	// setup
+	dclient := dynamicFake.NewSimpleDynamicClient(runtime.NewScheme())
+	cclient := apixFake.NewSimpleClientset(singleVersionFakeCRD)
+
+	cclient.Fake.PrependReactor("patch", "customresourcedefinitions", func(action k8stesting.Action) (handled bool, ret runtime.Object, err error) {
+		pa, ok := action.(k8stesting.PatchAction)
+		if !ok {
+			return true, nil, fmt.Errorf("not a patch action: %#v", action)
```

Comment:
We want to return `false` since we aren't handling this action - but given you're filtering using the `patch` verb in the `Prepend` call I would recommend just dropping this if block cause you're always going to get `PatchAction` type here

Reply:
Good catch. I've removed the unnecessary check too.